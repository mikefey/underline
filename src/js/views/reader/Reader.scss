.wrapper {
  height: 100%;
  position: absolute;
  width: 100%;
}

.content {
  height: 100%;
  overflow: hidden;
  position: absolute;
  width: 100%;
}

.bookContent {
  font-size: 20px;
  margin: 80px 0 0 20px;
  position: relative;
  width: 100%;
}

.bookContentDraggable {
  cursor: move;
  cursor: grab;
  cursor: -moz-grab;
  cursor: -webkit-grab;
  height: 100%;
  position: absolute;
  touch-action: auto;
  width: 100%;

  &:active {
    cursor: grabbing;
    cursor: -moz-grabbing;
    cursor: -webkit-grabbing;
  }

  iframe {
    font-size: 20px;
    line-height: 1.5em;
    pointer-events: none;
  }
}

.bookContentWrapper {
  float: left;
  position: relative;
  width: 100%;
}
