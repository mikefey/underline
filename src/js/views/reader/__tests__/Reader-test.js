import test from 'ava';
import React from 'react';
import { renderToString } from 'react-dom/server';

import Reader from './../Reader.jsx';

test('Reader component', (t) => {
  const readerState = {
    currentBookData: null,
    findingBookPosition: false,
  };

  const loaderState = {
    show: false,
    message: '',
  };

  const tableOfContentsState = {
    clickedButtonIndex: null,
    data: null,
    show: false,
  };

  const resizerState = {
    windowSize: {
      width: 1024,
      height: 768,
    },
  };

  const hideLoader = () => {};
  const showLoader = () => {};
  const loadBook = () => {};
  const showTableOfContents = () => {};
  const sliderReleased = () => {};
  const updateBookPositionData = () => {};
  const updateTableOfContents = () => {};
  const params = {};

  const tree = renderToString(
    <Reader
      readerState={readerState}
      loaderState={loaderState}
      tableOfContentsState={tableOfContentsState}
      hideLoader={hideLoader}
      showLoader={showLoader}
      loadBook={loadBook}
      params={params}
      resizerState={resizerState}
      showTableOfContents={showTableOfContents}
      sliderReleased={sliderReleased}
      updateBookPositionData={updateBookPositionData}
      updateTableOfContents={updateTableOfContents}
    />,
  );

  t.snapshot(tree);
});
