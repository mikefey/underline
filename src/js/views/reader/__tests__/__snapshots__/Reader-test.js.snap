// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reader component 1`] = `"<div class=\\"Reader__wrapper___1SShBfv-H4xY0jO\\" data-reactroot=\\"\\" data-reactid=\\"1\\" data-react-checksum=\\"841260369\\"><button class=\\"ArrowButton__wrapper___LW6dnzPENMQLGx1\\" data-reactid=\\"2\\"><div class=\\"ArrowButton__icon___3U6OyB1KlxCuEWN\\" data-reactid=\\"3\\"><div class=\\"ArrowButton__iconLine___16KAwsz8zR2PjhD\\" data-reactid=\\"4\\"></div><div class=\\"ArrowButton__iconLineTwo___3KOQrR_CEx1_mLb ArrowButton__iconLine___16KAwsz8zR2PjhD\\" data-reactid=\\"5\\"></div></div></button><div class=\\"Reader__content___1E3QY6EhKMpjk1p\\" data-reactid=\\"6\\"><div class=\\"Reader__bookContentDraggable___23xuXEX-G9FYUwF\\" style=\\"opacity:1;transform:translate3d(0px, 0, 0);transition-property:transform;transition-timing-function:cubic-bezier(1.000, 0.000, 0.000, 1.000);transition-duration:.3s;\\" data-reactid=\\"7\\"><div class=\\"PlaceholderPage__wrapper___27RHrfGqc8LL4o_ PlaceholderPage__left___3HBiybtJGG0Dfl9\\" data-reactid=\\"8\\"><div class=\\"PlaceholderPage__lineOne___3GueP9__dOz5GkC PlaceholderPage__line___1JI6E4CpHNdYb4G\\" data-reactid=\\"9\\"></div><div class=\\"PlaceholderPage__lineOne___3GueP9__dOz5GkC PlaceholderPage__line___1JI6E4CpHNdYb4G\\" data-reactid=\\"10\\"></div><div class=\\"PlaceholderPage__lineOne___3GueP9__dOz5GkC PlaceholderPage__line___1JI6E4CpHNdYb4G\\" data-reactid=\\"11\\"></div><div class=\\"PlaceholderPage__lineOne___3GueP9__dOz5GkC PlaceholderPage__line___1JI6E4CpHNdYb4G\\" data-reactid=\\"12\\"></div><div class=\\"PlaceholderPage__lineOne___3GueP9__dOz5GkC PlaceholderPage__line___1JI6E4CpHNdYb4G\\" data-reactid=\\"13\\"></div><div class=\\"PlaceholderPage__lineOne___3GueP9__dOz5GkC PlaceholderPage__line___1JI6E4CpHNdYb4G\\" data-reactid=\\"14\\"></div><div class=\\"PlaceholderPage__lineOne___3GueP9__dOz5GkC PlaceholderPage__line___1JI6E4CpHNdYb4G\\" data-reactid=\\"15\\"></div><div class=\\"PlaceholderPage__lineTwo___1jEzX07ljG9tzxL PlaceholderPage__line___1JI6E4CpHNdYb4G\\" data-reactid=\\"16\\"></div><div class=\\"PlaceholderPage__lineOne___3GueP9__dOz5GkC PlaceholderPage__line___1JI6E4CpHNdYb4G\\" data-reactid=\\"17\\"></div><div class=\\"PlaceholderPage__lineOne___3GueP9__dOz5GkC PlaceholderPage__line___1JI6E4CpHNdYb4G\\" data-reactid=\\"18\\"></div><div class=\\"PlaceholderPage__lineOne___3GueP9__dOz5GkC PlaceholderPage__line___1JI6E4CpHNdYb4G\\" data-reactid=\\"19\\"></div><div class=\\"PlaceholderPage__lineOne___3GueP9__dOz5GkC PlaceholderPage__line___1JI6E4CpHNdYb4G\\" data-reactid=\\"20\\"></div><div class=\\"PlaceholderPage__lineOne___3GueP9__dOz5GkC PlaceholderPage__line___1JI6E4CpHNdYb4G\\" data-reactid=\\"21\\"></div><div class=\\"PlaceholderPage__lineOne___3GueP9__dOz5GkC PlaceholderPage__line___1JI6E4CpHNdYb4G\\" data-reactid=\\"22\\"></div><div class=\\"PlaceholderPage__lineOne___3GueP9__dOz5GkC PlaceholderPage__line___1JI6E4CpHNdYb4G\\" data-reactid=\\"23\\"></div><div class=\\"PlaceholderPage__lineTwo___1jEzX07ljG9tzxL PlaceholderPage__line___1JI6E4CpHNdYb4G\\" data-reactid=\\"24\\"></div><div class=\\"PlaceholderPage__lineOne___3GueP9__dOz5GkC PlaceholderPage__line___1JI6E4CpHNdYb4G\\" data-reactid=\\"25\\"></div><div class=\\"PlaceholderPage__lineOne___3GueP9__dOz5GkC PlaceholderPage__line___1JI6E4CpHNdYb4G\\" data-reactid=\\"26\\"></div><div class=\\"PlaceholderPage__lineOne___3GueP9__dOz5GkC PlaceholderPage__line___1JI6E4CpHNdYb4G\\" data-reactid=\\"27\\"></div><div class=\\"PlaceholderPage__lineOne___3GueP9__dOz5GkC PlaceholderPage__line___1JI6E4CpHNdYb4G\\" data-reactid=\\"28\\"></div><div class=\\"PlaceholderPage__lineOne___3GueP9__dOz5GkC PlaceholderPage__line___1JI6E4CpHNdYb4G\\" data-reactid=\\"29\\"></div><div class=\\"PlaceholderPage__lineOne___3GueP9__dOz5GkC PlaceholderPage__line___1JI6E4CpHNdYb4G\\" data-reactid=\\"30\\"></div><div class=\\"PlaceholderPage__lineOne___3GueP9__dOz5GkC PlaceholderPage__line___1JI6E4CpHNdYb4G\\" data-reactid=\\"31\\"></div><div class=\\"PlaceholderPage__lineTwo___1jEzX07ljG9tzxL PlaceholderPage__line___1JI6E4CpHNdYb4G\\" data-reactid=\\"32\\"></div><div class=\\"PlaceholderPage__lineOne___3GueP9__dOz5GkC PlaceholderPage__line___1JI6E4CpHNdYb4G\\" data-reactid=\\"33\\"></div><div class=\\"PlaceholderPage__lineOne___3GueP9__dOz5GkC PlaceholderPage__line___1JI6E4CpHNdYb4G\\" data-reactid=\\"34\\"></div><div class=\\"PlaceholderPage__lineOne___3GueP9__dOz5GkC PlaceholderPage__line___1JI6E4CpHNdYb4G\\" data-reactid=\\"35\\"></div><div class=\\"PlaceholderPage__lineOne___3GueP9__dOz5GkC PlaceholderPage__line___1JI6E4CpHNdYb4G\\" data-reactid=\\"36\\"></div><div class=\\"PlaceholderPage__lineOne___3GueP9__dOz5GkC PlaceholderPage__line___1JI6E4CpHNdYb4G\\" data-reactid=\\"37\\"></div><div class=\\"PlaceholderPage__lineOne___3GueP9__dOz5GkC PlaceholderPage__line___1JI6E4CpHNdYb4G\\" data-reactid=\\"38\\"></div><div class=\\"PlaceholderPage__lineOne___3GueP9__dOz5GkC PlaceholderPage__line___1JI6E4CpHNdYb4G\\" data-reactid=\\"39\\"></div><div class=\\"PlaceholderPage__lineTwo___1jEzX07ljG9tzxL PlaceholderPage__line___1JI6E4CpHNdYb4G\\" data-reactid=\\"40\\"></div></div><div class=\\"Reader__bookContentWrapper___1X5DDxpjLR30crt\\" data-reactid=\\"41\\"><div class=\\"Reader__bookContent___354i32HLCfmSMtC\\" data-reactid=\\"42\\"></div></div><div class=\\"PlaceholderPage__wrapper___27RHrfGqc8LL4o_ PlaceholderPage__right___3qIOy2sN2laKWxg\\" data-reactid=\\"43\\"><div class=\\"PlaceholderPage__lineOne___3GueP9__dOz5GkC PlaceholderPage__line___1JI6E4CpHNdYb4G\\" data-reactid=\\"44\\"></div><div class=\\"PlaceholderPage__lineOne___3GueP9__dOz5GkC PlaceholderPage__line___1JI6E4CpHNdYb4G\\" data-reactid=\\"45\\"></div><div class=\\"PlaceholderPage__lineOne___3GueP9__dOz5GkC PlaceholderPage__line___1JI6E4CpHNdYb4G\\" data-reactid=\\"46\\"></div><div class=\\"PlaceholderPage__lineOne___3GueP9__dOz5GkC PlaceholderPage__line___1JI6E4CpHNdYb4G\\" data-reactid=\\"47\\"></div><div class=\\"PlaceholderPage__lineOne___3GueP9__dOz5GkC PlaceholderPage__line___1JI6E4CpHNdYb4G\\" data-reactid=\\"48\\"></div><div class=\\"PlaceholderPage__lineOne___3GueP9__dOz5GkC PlaceholderPage__line___1JI6E4CpHNdYb4G\\" data-reactid=\\"49\\"></div><div class=\\"PlaceholderPage__lineOne___3GueP9__dOz5GkC PlaceholderPage__line___1JI6E4CpHNdYb4G\\" data-reactid=\\"50\\"></div><div class=\\"PlaceholderPage__lineTwo___1jEzX07ljG9tzxL PlaceholderPage__line___1JI6E4CpHNdYb4G\\" data-reactid=\\"51\\"></div><div class=\\"PlaceholderPage__lineOne___3GueP9__dOz5GkC PlaceholderPage__line___1JI6E4CpHNdYb4G\\" data-reactid=\\"52\\"></div><div class=\\"PlaceholderPage__lineOne___3GueP9__dOz5GkC PlaceholderPage__line___1JI6E4CpHNdYb4G\\" data-reactid=\\"53\\"></div><div class=\\"PlaceholderPage__lineOne___3GueP9__dOz5GkC PlaceholderPage__line___1JI6E4CpHNdYb4G\\" data-reactid=\\"54\\"></div><div class=\\"PlaceholderPage__lineOne___3GueP9__dOz5GkC PlaceholderPage__line___1JI6E4CpHNdYb4G\\" data-reactid=\\"55\\"></div><div class=\\"PlaceholderPage__lineOne___3GueP9__dOz5GkC PlaceholderPage__line___1JI6E4CpHNdYb4G\\" data-reactid=\\"56\\"></div><div class=\\"PlaceholderPage__lineOne___3GueP9__dOz5GkC PlaceholderPage__line___1JI6E4CpHNdYb4G\\" data-reactid=\\"57\\"></div><div class=\\"PlaceholderPage__lineOne___3GueP9__dOz5GkC PlaceholderPage__line___1JI6E4CpHNdYb4G\\" data-reactid=\\"58\\"></div><div class=\\"PlaceholderPage__lineTwo___1jEzX07ljG9tzxL PlaceholderPage__line___1JI6E4CpHNdYb4G\\" data-reactid=\\"59\\"></div><div class=\\"PlaceholderPage__lineOne___3GueP9__dOz5GkC PlaceholderPage__line___1JI6E4CpHNdYb4G\\" data-reactid=\\"60\\"></div><div class=\\"PlaceholderPage__lineOne___3GueP9__dOz5GkC PlaceholderPage__line___1JI6E4CpHNdYb4G\\" data-reactid=\\"61\\"></div><div class=\\"PlaceholderPage__lineOne___3GueP9__dOz5GkC PlaceholderPage__line___1JI6E4CpHNdYb4G\\" data-reactid=\\"62\\"></div><div class=\\"PlaceholderPage__lineOne___3GueP9__dOz5GkC PlaceholderPage__line___1JI6E4CpHNdYb4G\\" data-reactid=\\"63\\"></div><div class=\\"PlaceholderPage__lineOne___3GueP9__dOz5GkC PlaceholderPage__line___1JI6E4CpHNdYb4G\\" data-reactid=\\"64\\"></div><div class=\\"PlaceholderPage__lineOne___3GueP9__dOz5GkC PlaceholderPage__line___1JI6E4CpHNdYb4G\\" data-reactid=\\"65\\"></div><div class=\\"PlaceholderPage__lineOne___3GueP9__dOz5GkC PlaceholderPage__line___1JI6E4CpHNdYb4G\\" data-reactid=\\"66\\"></div><div class=\\"PlaceholderPage__lineTwo___1jEzX07ljG9tzxL PlaceholderPage__line___1JI6E4CpHNdYb4G\\" data-reactid=\\"67\\"></div><div class=\\"PlaceholderPage__lineOne___3GueP9__dOz5GkC PlaceholderPage__line___1JI6E4CpHNdYb4G\\" data-reactid=\\"68\\"></div><div class=\\"PlaceholderPage__lineOne___3GueP9__dOz5GkC PlaceholderPage__line___1JI6E4CpHNdYb4G\\" data-reactid=\\"69\\"></div><div class=\\"PlaceholderPage__lineOne___3GueP9__dOz5GkC PlaceholderPage__line___1JI6E4CpHNdYb4G\\" data-reactid=\\"70\\"></div><div class=\\"PlaceholderPage__lineOne___3GueP9__dOz5GkC PlaceholderPage__line___1JI6E4CpHNdYb4G\\" data-reactid=\\"71\\"></div><div class=\\"PlaceholderPage__lineOne___3GueP9__dOz5GkC PlaceholderPage__line___1JI6E4CpHNdYb4G\\" data-reactid=\\"72\\"></div><div class=\\"PlaceholderPage__lineOne___3GueP9__dOz5GkC PlaceholderPage__line___1JI6E4CpHNdYb4G\\" data-reactid=\\"73\\"></div><div class=\\"PlaceholderPage__lineOne___3GueP9__dOz5GkC PlaceholderPage__line___1JI6E4CpHNdYb4G\\" data-reactid=\\"74\\"></div><div class=\\"PlaceholderPage__lineTwo___1jEzX07ljG9tzxL PlaceholderPage__line___1JI6E4CpHNdYb4G\\" data-reactid=\\"75\\"></div></div></div><div class=\\"LocationSlider__wrapper___25I5ZQj0-ltKOtx\\" data-reactid=\\"76\\"><div class=\\"LocationSlider__sliderWrapper___3lzgP5ER0X9BBn3\\" data-reactid=\\"77\\"><div class=\\"LocationSlider__sliderTrack___23RWsOlNN2vnaPh\\" data-reactid=\\"78\\"></div><div class=\\"LocationSlider__sliderProgress___3PmqidtlheKQTSB LocationSlider__sliderTrack___23RWsOlNN2vnaPh\\" style=\\"width:0px;\\" data-reactid=\\"79\\"></div><div class=\\"LocationSlider__sliderDraggable___36FLOdYP5cCi8Kx\\" style=\\"transform:translate3d(0px, 0, 0);\\" data-reactid=\\"80\\"><div class=\\"LocationSlider__sliderHandle___1fsJj3qVXHKa7Wf\\" data-reactid=\\"81\\"></div></div></div></div><div class=\\"Header__wrapper___2PTLf_nkW7S2HEU\\" data-reactid=\\"82\\"><div class=\\"Header__content___2VIdu565Foup5Pb\\" data-reactid=\\"83\\"><a class=\\"BackButton__wrapper___3allv248BLEnfXA\\" data-reactid=\\"84\\"><div class=\\"BackButton__icon___302Gb-yIGqd5oDJ\\" data-reactid=\\"85\\"><div class=\\"BackButton__iconLineOne___3FLnuJCdoUNaVak\\" data-reactid=\\"86\\"></div><div class=\\"BackButton__iconLineTwo___24Pm7a3li3bBJOC\\" data-reactid=\\"87\\"></div><div class=\\"BackButton__iconLineThree___1ES8_m0XRvJqRDJ\\" data-reactid=\\"88\\"></div></div></a><h2 class=\\"Header__title___1_ebCiqkttbS44m\\" data-reactid=\\"89\\"></h2><button class=\\"OpenTOCButton__wrapper___JVDX5q0UrnRPYR_\\" data-reactid=\\"90\\"><div class=\\"OpenTOCButton__icon___AntlYF_8YmZKxmI\\" data-reactid=\\"91\\"><div class=\\"OpenTOCButton__iconLine___3LMQ5Yls8vNIqiT\\" data-reactid=\\"92\\"></div><div class=\\"OpenTOCButton__iconLine___3LMQ5Yls8vNIqiT\\" data-reactid=\\"93\\"></div><div class=\\"OpenTOCButton__iconLine___3LMQ5Yls8vNIqiT\\" data-reactid=\\"94\\"></div></div></button></div></div></div><button class=\\"ArrowButton__wrapperRight___41iaXmNw7L2cXrY ArrowButton__wrapper___LW6dnzPENMQLGx1\\" data-reactid=\\"95\\"><div class=\\"ArrowButton__icon___3U6OyB1KlxCuEWN\\" data-reactid=\\"96\\"><div class=\\"ArrowButton__iconLine___16KAwsz8zR2PjhD\\" data-reactid=\\"97\\"></div><div class=\\"ArrowButton__iconLineTwo___3KOQrR_CEx1_mLb ArrowButton__iconLine___16KAwsz8zR2PjhD\\" data-reactid=\\"98\\"></div></div></button></div>"`;
